crates/driver/src/tests/cases/protocol_fees.rs

```diff
@@ -8,46 +8,66 @@ use crate::{
             ab_liquidity_quote,
             ab_order,
             ab_solution,
+            fee::{Policy, Quote},
             ExpectedOrderAmounts,
-            FeePolicy,
             Test,
         },
     },
 };
 
+struct Amounts {
+    sell: eth::U256,
+    buy: eth::U256,
+}
+
+struct Execution {
+    // The executed net-amounts (including network fee) reported by the solver
+    solver: Amounts,
+    // The executed net-amounts (including network and protocol) reported by the driver
+    driver: Amounts,
+}
+
+struct Order {
+    amounts: Amounts,
+    side: order::Side,
+}
+
 struct TestCase {
-    order_side: order::Side,
-    fee_policy: FeePolicy,
-    order_sell_amount: eth::U256,
-    solver_fee: Option<eth::U256>,
-    quote_sell_amount: eth::U256,
-    quote_buy_amount: eth::U256,
-    executed: eth::U256,
-    executed_sell_amount: eth::U256,
-    executed_buy_amount: eth::U256,
+    order: Order,
+    fee_policy: Policy,
+    execution: Execution,
 }
 
 async fn protocol_fee_test_case(test_case: TestCase) {
     let test_name = format!(
         "Protocol Fee: {:?} {:?}",
-        test_case.order_side, test_case.fee_policy
+        test_case.order.side, test_case.fee_policy
     );
+    // Adjust liquidity pools so that the order is executable at the amounts
+    // expected from the solver.
     let quote = ab_liquidity_quote()
-        .sell_amount(test_case.quote_sell_amount)
-        .buy_amount(test_case.quote_buy_amount);
+        .sell_amount(test_case.execution.solver.sell)
+        .buy_amount(test_case.execution.solver.buy);
     let pool = ab_adjusted_pool(quote);
+
+    // Amounts expected to be returned by the driver after fee processing
     let expected_amounts = ExpectedOrderAmounts {
-        sell: test_case.executed_sell_amount,
-        buy: test_case.executed_buy_amount,
+        sell: test_case.execution.driver.sell,
+        buy: test_case.execution.driver.buy,
     };
+
     let order = ab_order()
         .kind(order::Kind::Limit)
-        .sell_amount(test_case.order_sell_amount)
-        .side(test_case.order_side)
-        .solver_fee(test_case.solver_fee)
+        .sell_amount(test_case.order.amounts.sell)
+        .buy_amount(test_case.order.amounts.buy)
+        // Expected amounts already account for network fee, so it doesn't matter for the math.
+        // However, it cannot be zero, otherwise the order would be perceived as a StaticFee orders (which cannot have Protocol Fees)
```

Comment:
nit: todo clean this up as part of the https://github.com/cowprotocol/services/issues/2507